nubis/terraform/userdata.tf

```diff
@@ -1,95 +1,71 @@
-    CONSUL_ACL_TOKEN    = "${var.consul_acl_token}"
-    NUBIS_PURPOSE       = "${var.purpose}-master"
-    NUBIS_ENVIRONMENT   = "${var.environment}"
-    NUBIS_ARENA         = "${var.arena}"
-    NUBIS_DOMAIN        = "${var.nubis_domain}"
-    NUBIS_ACCOUNT       = "${var.account}"
-    NUBIS_STACK         = "${var.service_name}-${var.environment}"
-    NUBIS_SUDO_GROUPS   = "${var.nubis_sudo_groups}"
-    NUBIS_USER_GROUPS   = "${var.nubis_user_groups}"
-    NUBIS_SWAP_SIZE_MEG = 0
-  }
+module "userdata_master" {
+  source            = "github.com/nubisproject/nubis-terraform//worker/userdata?ref=develop"
+  region            = "${var.region}"
+  nubis_domain      = "${var.nubis_domain}"
+  arena             = "${var.arena}"
+  environment       = "${var.environment}"
+  account           = "${var.account}"
+  service_name      = "${var.service_name}"
+  swap_size_meg     = "0"
+  nubis_user_groups = "${var.nubis_user_groups}"
+  nubis_sudo_groups = "${var.nubis_sudo_groups}"
+  consul_token      = "${var.consul_acl_token}"
+  purpose           = "master"
 }
 
-data "template_file" "user_data_cloudconfig_master" {
-  #count    = "${var.enabled}"
-  template = "${file("${path.module}/templates/userdata_cloudconfig.tpl")}"
+data "template_file" "userdata_master" {
+  template = "${file("${path.module}/templates/additionnal_userdata.tpl")}"
 
   vars {
     NAME    = "nubis-metadata"
-    PAYLOAD = "${base64encode(data.template_file.nubis_metadata_master.rendered)}"
-
-    # The nubis-metadata script looks here for it
-    LOCATION = "/var/cache/nubis/userdata"
+    PAYLOAD = "${indent(6, module.userdata_master.cloudinit)}"
   }
 }
 
-data "template_file" "nubis_metadata_node" {
-  #count    = "${var.enabled}"
-  template = "${file("${path.module}/templates/userdata.tpl")}"
-
-  vars {
-    NUBIS_PROJECT       = "${var.service_name}-node"
-    CONSUL_ACL_TOKEN    = "${var.consul_acl_token}"
-    NUBIS_PURPOSE       = "${var.purpose}-node"
-    NUBIS_ENVIRONMENT   = "${var.environment}"
-    NUBIS_ARENA         = "${var.arena}"
-    NUBIS_DOMAIN        = "${var.nubis_domain}"
-    NUBIS_ACCOUNT       = "${var.account}"
-    NUBIS_STACK         = "${var.service_name}-${var.environment}"
-    NUBIS_SUDO_GROUPS   = "${var.nubis_sudo_groups}"
-    NUBIS_USER_GROUPS   = "${var.nubis_user_groups}"
-    NUBIS_SWAP_SIZE_MEG = 0
-  }
+module "userdata_minion" {
+  source            = "github.com/nubisproject/nubis-terraform//worker/userdata?ref=develop"
+  region            = "${var.region}"
+  nubis_domain      = "${var.nubis_domain}"
+  arena             = "${var.arena}"
+  environment       = "${var.environment}"
+  account           = "${var.account}"
+  service_name      = "${var.service_name}"
+  swap_size_meg     = "0"
+  nubis_user_groups = "${var.nubis_user_groups}"
+  nubis_sudo_groups = "${var.nubis_sudo_groups}"
+  consul_token      = "${var.consul_acl_token}"
+  purpose           = "minion"
```

Comment:
Can we not name this `minion`? I think `minion` is an old moniker and its called node these days

Reply:
Ah, sure thing! Thought minion was the current name